packages/carbon/src/classes/Client.ts

```diff
@@ -162,129 +115,87 @@ export class Client {
 					this.modalHandler.registerModal(new modal())
 			}
 		}
-		if (this.options.autoDeploy) this.deployCommands()
-		this.setupRoutes()
-	}
-
-	/**
-	 * Deploy the commands registered to Discord.
-	 * This is automatically called when running in NodeJS mode.
-	 */
-	async deployCommands() {
-		try {
-			const commands = this.commands
-				.filter((x) => x.name !== "*")
-				.map((command) => {
-					return command.serialize()
-				})
-			await this.rest.put(Routes.applicationCommands(this.options.clientId), {
-				body: commands
-			})
-			console.log(`Deployed ${commands.length} commands to Discord`)
-		} catch (err) {
-			console.error("Failed to deploy commands")
-			console.error(err)
+		if (this.options.autoDeploy) {
+			// If this ever requires req/ctx, this will need to change
+			this.handleDeployCommandsRequest()
 		}
 	}
 
-	/**
-	 * Setup the routes for the client
-	 */
-	private setupRoutes() {
-		this.router.get("/", () => {
-			if (this.options.redirectUrl)
-				return Response.redirect(this.options.redirectUrl, 302)
-			throw new StatusError(404)
+	private appendRoutes() {
+		this.routes.push({
+			method: "GET",
+			path: "/deploy",
+			handler: this.handleDeployCommandsRequest.bind(this)
 		})
```

Comment:
Can we make enabling this a config option on the client, so that users are able to entirely not have that route if they don't want it and deploy another way manually instead. Spamming it would easily be a way to ratelimit the bot (though adding the protection will also help with that)

Reply:
Will do, should all routes be enableable/disableable as well?

Reply:
Up to you tbh. Maybe make the required ones like /interaction always be there, but extra ones like /deploy be optional?

Reply:
Okay I'll make everyone enableable/disableable, I can think of scenarios where someone want might to enable /deploy but disable /interaction. I guess I'll make every route enabled by default (will add protections to deploy route beforehand) and add options to disable individual routes.